src/com/projectkorra/projectkorra/earthbending/passive/DensityShift.java

```diff
@@ -37,6 +37,8 @@ public static boolean softenLanding(final Player player) {
 		} else if (bPlayer.canMetalbend() && ElementalAbility.isMetalBlock(block)) {
 			return true;
 		}
+		if ((player.getLocation().getY() % 1) != 0)
+			player.teleport(player.getLocation().add(0, 0.1, 0));
```

Comment:
Instead of teleporting the player up 0.1, why not teleport them to `new Location(player.getWorld(), player.getLocation().getX(), player.getLocation().getBlock().getRelative(BlockFace.UP).getY(), player.getLocation().getZ())` or something along those lines.

Reply:
also would work

Reply:
This would be much better and more modular.